mongo/dimension_store.go

```diff
@@ -85,13 +85,20 @@ func (m *Mongo) GetDimensions(datasetID, versionID string) ([]bson.M, error) {
 	return results, nil
 }
 
-// GetDimensionOptions returns all dimension options for a dimensions within a dataset.
-func (m *Mongo) GetDimensionOptions(version *models.Version, dimension string) (*models.DimensionOptionResults, error) {
+// GetDimensionOptions returns dimension options for a dimensions within a dataset, according to the provided limit and offest.
+// Offset and limit will not be validated, assuming they are positive or zero. Zero limit is equivalent to no limit (all items starting at offset will be returned)
+func (m *Mongo) GetDimensionOptions(version *models.Version, dimension string, offset, limit int) (*models.DimensionOptionResults, error) {
```

Comment:
Consider adding additional validation for `offset` and `limit` here - I know you've already do this in the handler but it's not unreasonable for this method to be used somewhere else as well.